gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp

```diff
@@ -289,7 +288,9 @@ OGRGeometryCollection *GeometryReader::readGeometryCollection()
     if (parts == nullptr)
         return CPLErrorInvalidPointer("parts data");
     auto gc = new OGRGeometryCollection();
-    for (uoffset_t i = 0; i < parts->Length(); i++) {
+    if (parts == nullptr)
```

Comment:
this test has been done just above

Reply:
Perhaps it was meant to check the gc. By the way, the gc could be captured with unique_ptr for exception safety

Reply:
> Perhaps it was meant to check the gc

as gc results from a new, the only failure mode would be a C++ exception, not nullptr, right ?

Reply:
Yes, correct, but there is chance author's intention was to check that anyway.

Reply:
Ah no it's a mistake when I merged locally.